src/modules/75-cf/pages/onboarding/__test__/OnboardingDetailPage.test.tsx

```diff
@@ -30,6 +33,8 @@ const renderComponent = (): RenderResult => {
   )
 }
 
+jest.mock('framework/PreferenceStore/PreferenceStoreContext')
```

Comment:
Instead of mocking this and then mocking the implementation of the hook in each test, try just using a single mock or spy.

```suggestion
jest.mock('framework/PreferenceStore/PreferenceStoreContext', () => {
  usePreferenceStore: jest.fn().mockReturnValue({
    setPreference: jest.fn(),
    preference: true,
    clearPreference: jest.fn()
  })
})
```

or using a spy
```suggestion
import * as preferenceStoreContext from 'framework/PreferenceStore/PreferenceStoreContext'

// ...

jest.spyOn(preferenceStoreContext, 'usePreferenceStore').mockReturnValue({
  setPreference: jest.fn(),
  preference: true,
  clearPreference: jest.fn()
})

```

Reply:
Thanks, I have used the spy :) I was copying how we did it in other modules but this is far better